index.html

```diff
@@ -0,0 +1,578 @@
+Versions can be ordered with the following pseudocode:
+
+```typescript
+compareVersions(a: Version, b: Version) -> LESS_THAN | EQUAL | GREATER_THAN {
+  if a.major < b.major {
+    return LESS_THAN
+  }
+  if a.major == b.major {
+    if a.minor == b.minor { return EQUAL }
+    if a.minor > b.minor { return GREATER_THAN }
+    return LESS_THAN
+  }
+  if a.major > b.major {
+    return GREATER_THAN
+  }
+}
+```
+
+Note that this ordering does *not* guarantee that if A > B then A can satisfy a version requirement of B. For example, `2.0 > 1.9`, but `2.0` cannot [satisfy](#satisfying-version-requirements) the requirements of a document which references version `1.9`, since the major version update introduced possibly-breaking changes.
+
+### Satisfaction
+
+Given a version `requested` by a document and an `available` version of an implementation, the following pseudocode will determine if the `available` version can satisfy the `requested` version:
+
+```typescript
+canSatisfy(requested: Version, available: Version) =
+  requested.major == available.major &&
+  (requested.major > 0 && (requested.minor <= available.minor)
+    || (requested.minor == available.minor))
+```
+
+### Referencing versions and activating implementations
+
+Schema documents **must** reference a spec version which supports all the schema elements and behaviors required by the document. As a practical matter, authors should prefer to reference the [lowest](#ordering) versions which can [satisfy](#satisfaction) their requirements, as this imposes the fewest constraints on processors and is thus more likely to be supported.
+
+For every spec they support, processors **must** activate an implementation which can [satisfy](#satisfaction) the version required by the document.
+
+### Backus-Naur Form for versions and version tags
+
+```plaintext
+#::[BNF grammar for SemVer specifiers]
+‹version tag› ::= "v" ‹version›
+
+‹version› ::= ‹major›
+            | ‹major› "." ‹minor›
+
+‹major› ::= ‹numeric identifier›
+‹minor› ::= ‹numeric identifier›
+‹numeric identifier› ::= "0"
+                       | ‹positive digit›
+                       | ‹positive digit› ‹digits›
+‹digits› ::= ‹digit›
+           | ‹digit› ‹digits›
+
+‹digit› ::= "0"
+          | ‹positive digit›
+‹positive digit› ::= "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9"
+```
+
+## Extensibility
+
+All directives, scalars, enums, input objects, and other schema elements defined by specs are **extensible**. This means that their definitions within a core schema document do not have to exactly match definitions provided by specs. Instead, definitions should match all usages within the document.
+
+Standard extensibility rules:
+- **Directives** may be defined as taking additional arguments, provided those arguments are [prefixed](#prefixing) with the name of a spec followed by two underscores ("`__`"). These arguments are interpreted as metadata-on-metadata, and are routed to the declaring spec for interpretation
```

Comment:
So this explicitly means you can't just add your arguments to spec-defined directives in your graph's schema without writing and linking in a spec for them. Is that intentional? I can see arguments in either direction.

Reply:
One thing we need to do is define the core validations, one of which checks that every name is either bound to machinery or to the export schema. Arbitrary new arguments on directives would fail that validation, but would succeed if you included a `@core(using:)` line that brings in the namespace. So basically yes, you have to be `using: ".../mysomething/v1.0"` to add `mysomething__somearg` arguments to a directive (or input object).

It is technically non-compliant to be `using` a URL that doesn't go anywhere, but as a practical matter, no (currently-defined) automated process is trying to read those URLs, so you can really just make one up and use it to grab a namespace.

As we begin to look to a future where there _is_ a machine-readable form of specs which automated tools _do_ load from the URL, we might consider introducing a way to grab a namespace without specifying what's in it. Then you could use that namespace for `namespaced__arguments` and it would pass validation.

(Also, in such a future, it's very likely that the purely-hypothetical-at-present compiler would consult the local working tree first for files that are `@spec(specifying: "http://whatever/spec/url...")` before trying to fetch the spec schema remotely. So even if we didn't have an escape hatch that let you declare namespaces without specifying them, specifying them wouldn't necessarily require actually deploying something to the web.)